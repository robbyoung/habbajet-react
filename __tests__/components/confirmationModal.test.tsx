import React from 'react';
import renderer from 'react-test-renderer';
import ConfirmationModal from '../../app/components/confirmationModal';
import {render, fireEvent, wait} from '@testing-library/react-native';

const mockDismissModal = jest.fn();
jest.mock('react-native-navigation', () => ({
    Navigation: {
        dismissModal: (id: string) => mockDismissModal(id),
    },
}));

describe('Confirmation Modal Component', () => {
    it('can render a confirmation modal', () => {
        const component = renderer.create(
            <ConfirmationModal
                text="This is a confirmation modal"
                id="testId"
                onConfirm={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will run the onConfirm callback if the yes button is pressed', async () => {
        const onConfirm = jest.fn();
        const {getByTestId} = render(
            <ConfirmationModal
                text="This is a confirmation modal"
                id="testId"
                onConfirm={onConfirm}
            />,
        );

        const row = getByTestId('button-yes');
        fireEvent.press(row);
        await wait(() => expect(onConfirm).toBeCalled());
    });

    it('will run Navigation.dismissModal if the no button is pressed', async () => {
        const {getByTestId} = render(
            <ConfirmationModal
                text="This is a confirmation modal"
                id="testId"
                onConfirm={() => undefined}
            />,
        );

        const button = getByTestId('button-no');
        fireEvent.press(button);
        await wait(() => expect(mockDismissModal).toBeCalledWith('testId'));
    });

    it('will run Navigation.dismissModal if the background is pressed', async () => {
        const {getByTestId} = render(
            <ConfirmationModal
                text="This is a confirmation modal"
                id="testId"
                onConfirm={() => undefined}
            />,
        );

        const button = getByTestId('modal-background');
        fireEvent.press(button);
        await wait(() => expect(mockDismissModal).toBeCalledWith('testId'));
    });
});
