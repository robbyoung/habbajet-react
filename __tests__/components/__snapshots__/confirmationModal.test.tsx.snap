// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Modal Component can render a confirmation modal 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#95959599",
      "height": "100%",
      "width": "100%",
    }
  }
  testID="modal-background"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 20,
        "marginLeft": "15%",
        "marginTop": "20%",
        "padding": 20,
        "width": "70%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Abel",
          "fontSize": 24,
          "paddingBottom": 20,
          "textAlign": "center",
        }
      }
    >
      This is a confirmation modal
    </Text>
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#959595",
            "borderRadius": 10,
            "marginBottom": 10,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="button-yes"
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontFamily": "Abel",
                "fontSize": 30,
                "padding": 10,
                "textAlign": "center",
                "width": "100%",
              },
            ]
          }
        >
          Yes
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#959595",
            "borderRadius": 10,
            "marginBottom": 10,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="button-no"
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontFamily": "Abel",
                "fontSize": 30,
                "padding": 10,
                "textAlign": "center",
                "width": "100%",
              },
            ]
          }
        >
          No
        </Text>
      </View>
    </View>
  </View>
</View>
`;
